cfme/base/login.py

```diff
@@ -51,35 +73,29 @@ def current_fullname(self):
 
     @property
     def current_groupname(self):
-        # TODO: Move these locators and accessors to a widget "Group Dropdown" widget
         # 5.9 Locators for finding current group when user has multiple groups
-        current_group_locator = '{}/ul/li/a[@title="Currently Selected Group"]'.format(
-            self.group_list_locator)
         current_group_marker = ' (Current Group)'
-        try:
-            current_group = self.browser.element(current_group_locator)
-            return self.browser.text(current_group).replace(current_group_marker, '')
-        except NoSuchElementException:
-            return self.settings.items[1].strip()
+        current_groups = self.settings.groups.items
+
+        for group in current_groups:
+            if current_group_marker in group:
+                return group.replace(current_group_marker, '')
+        else:
```

Comment:
It is correct. In this context it's probably unnecessary since it's functions the same as replacing with a same return statement at the end of the method. I reworked it to make it clearer that we return the current group when only one group is assigned